impeller/renderer/frame_captor.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/macros.h"
+
+namespace impeller {
+
+//------------------------------------------------------------------------------
+/// @brief      Frame capture configuration.
+///
+struct FrameCaptorConfiguration {
+  /// This param is for metal backend.
+  /// When this value is true, a gpu trace file will be saved in devices when
+  /// capture finishes. Otherwise, the Xcode will automatically open and show
+  /// trace result.
+  ///
+  bool mtlSaveGPUTraceDocument = false;
```

Comment:
This header is backend agnostic so I am not sure how best to reconcile the Metal specificity of this option. Perhaps just a `MetalConfiguration` sub-struct? Also, please use the style guide for naming `save_gpu_trace_as_document`.

Reply:
Thx for reviewing this^_^

This is what I think about:
In this config, we gather all backends gpu trace params. And every backend just picks the corresponding params to use. 

Such as...
/// for metal backends
bool xxx_mtl1 = false
bool xxx_mtl2 = false

/// for gl backends
bool xxx_gl = false
bool xxx_gl = false

/// for vulkan backends
................